dbcrossbarlib/src/tokio_glue.rs

```diff
@@ -191,9 +249,9 @@ impl SyncStreamWriter {
     #[allow(dead_code)]
     pub(crate) fn send_error(&mut self, err: Error) -> io::Result<()> {
         debug!(self.ctx.log(), "sending error: {}", err);
-        if let Some(sender) = self.sender.take() {
-            match sender.send(Err(err)).wait() {
-                Ok(sender) => {
+        if let Some(mut sender) = self.sender.take() {
```

Comment:
This whole `self.sender.take()` business is suspicious, now that we have the new APIs.